packages/root/src/Root.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { HTMLElementProps } from '@leafygreen-ui/lib';
+
+type ButtonProps<T> = HTMLElementProps<'button'> &
+  T & {
+    href?: undefined;
+  };
+
+type Anchor<T> = HTMLElementProps<'a'> &
+  T & {
+    href: string;
+  };
+
+type CustomElementProps<T> = T & {
+  as: React.ElementType<any>;
+  [key: string]: any;
+};
+
+type RootProps<T> = ButtonProps<T> | Anchor<T> | CustomElementProps<T>;
+
+function usesCustomElement<T>(
+  props: RootProps<T>,
+): props is CustomElementProps<T> {
+  return (props as any).as != null;
+}
+
+function usesLinkElement<T>(props: RootProps<T>): props is Anchor<T> {
+  return props.href != null;
+}
+
+/**
+ * # Root
+ *
+ * Root component
+ *
+ * ```
+<Root href="https://mongodb.design">Anchors Away!</Root>
+```
+ * @param props.children Takes priority in determining what tag the `<Root />` element will render.
+ * @param props.href Determines the location that an anchor tag will point to. If prop is set without the `as` prop, `<Root />` will render `a` tags.
+ * @param props.as Callback to change the open state of the Menu.
```

Comment:
The `@param props.children` and `@param props.as` are incorrect here. I would also update these to match `README.md` exactly whenever possible.

Reply:
Ah thank you -- oversight when c/ping